### ISSUE TITLE
Chrome windows vanishing

### ISSUE THREAD

**Issue by csis0247** [deanoemcke/thegreatsuspender#1071]
* Extension version: Whatever version as of  2019-11-05.
 * Browser name & version: Chrome Canary, Dev, and Beta as of 2019-11-05
 * Operating system & version: Windows 10 Home
* "Apply Chrome's built-in memory-saving when suspending" enabled.

This issue has existed for a while, meaning, from at least October, but only for development builds of Chrome. I thought the devs were handling this so I didn't pay much attention. But today, a similar (but not identical) issue #1070 happened in normal Chrome, so I feel obliged to report this previous one.

As shown in the screenshot below, from last month, The Great Suspender reported 8 windows, but only 6 were open. Those windows were hidden, meaning that they were not physically opened in Windows, but the tabs inside were real user-opened tabs. I could restore them via "open and suspend", but that would mean that I would get duplicated windows in The Great Suspender's record. But still, it was usable.

![2019-11-05 The Great Suspender is losing windows](https://user-images.githubusercontent.com/21369255/70851046-eab3d780-1e90-11ea-91a2-a76a26aeb196.png)

Today, however, in normal Chrome, it was not just windows that went hidden, but actually losing windows. That would be bug report #1070 .

**Comment by bejl** [comment]
Same thing happened to me few days ago, extension held about 80 tabs in chromes memory without showing it and I couldn't find the way to restore tabs. 

Last versions of chrome, extension and WIN10 pro. 

The weird part is when I remove extension from chrome and install it again it keeps showing the hidden tabs in chrome's task manager.

**Comment by Infinitay** [comment]
Same thing is happening for me but I think it started recently because I only noticed it when **only one of my 3 windows** were showing up, but Great Suspender always reported that _they were open_, and started _duplicating_ after each time I manually closed all my tabs and resuspended just those 3 windows.

[Here is what I see with The Great Suspender's Session Management page](https://user-images.githubusercontent.com/6964154/70927998-0ab4d980-1ffe-11ea-9f63-e674b9d66648.png)

[Here is what I see in the debugger page for The Great Suspender](https://user-images.githubusercontent.com/6964154/70928236-94fd3d80-1ffe-11ea-9338-8050788e2e69.png)

<details>
<summary>Chrome Console Output</summary>
<br>

> gsUtils.js:41 505 "12:18:00" "send message to tab" Object
12:18:00.018 gsUtils.js:41 505 "12:18:00" "response from tab" Object
12:18:00.018 gsUtils.js:41 505 "12:18:00" "Content script status: formInput"
12:18:00.018 gsUtils.js:41 505 "12:18:00" "Focused tab status: formInput"
12:18:00.519 gsUtils.js:41 505 "12:18:00" "new stationary tab focus handled"
12:18:07.711 gsUtils.js:41 508 "12:18:07" "tab gained focus"
12:18:07.715 gsUtils.js:41 508 "12:18:07" "Focused tab status: special"
12:18:08.216 gsUtils.js:41 508 "12:18:08" "new stationary tab focus handled"
12:18:08.217 gsUtils.js:41 505 "12:18:08" "Removing tab timer."
12:18:08.217 gsUtils.js:41 505 "12:18:08" "Adding tab timer for: Mon Dec 16 2019 14:18:08 GMT-0500 (Eastern Standard Time)"
12:18:15.536 gsUtils.js:41 505 "12:18:15" "tab gained focus"
12:18:15.544 gsUtils.js:41 505 "12:18:15" "send message to tab" Object
12:18:15.547 gsUtils.js:41 505 "12:18:15" "response from tab" Object
12:18:15.547 gsUtils.js:41 505 "12:18:15" "Content script status: formInput"
12:18:15.547 gsUtils.js:41 505 "12:18:15" "Focused tab status: formInput"
12:18:15.951 gsUtils.js:41 508 "12:18:15" "tab gained focus"
12:18:15.958 gsUtils.js:41 508 "12:18:15" "Focused tab status: special"
12:18:16.459 gsUtils.js:41 508 "12:18:16" "new stationary tab focus handled"
12:18:16.997 gsUtils.js:41 511 "12:18:16" "tab created. tabUrl: "
12:18:16.998 gsUtils.js:41 511 "12:18:16" "tab gained focus"
12:18:17.001 gsUtils.js:41 511 "12:18:17" "send message to tab" Object
12:18:17.005 gsUtils.js:41 511 "12:18:17" "response from tab" undefined
12:18:17.006 gsUtils.js:54 WARNING: 511 12:18:17 Failed to getContentScriptStatus Object 
    at chrome-extension://klbibkeccnjlkjkiokjodocebajanakg/js/background.js:1283:21
12:18:17.006 gsUtils.js:41 511 "12:18:17" "checkQueue" "Queueing tab for responsiveness check."
12:18:17.021 gsUtils.js:41 511 "12:18:17" "tab url changed. changeInfo: " Object
12:18:17.060 gsUtils.js:41 511 "12:18:17" "checkQueue" "Executing executorFn for tab."
12:18:17.061 gsUtils.js:41 511 "12:18:17" "send message to tab" Object
12:18:17.063 gsUtils.js:41 511 "12:18:17" "response from tab" undefined
12:18:17.063 gsUtils.js:41 511 "12:18:17" "checkQueue" "Tab is not responding but active. Will wait for potential auto reload."
12:18:17.063 gsUtils.js:41 511 "12:18:17" "checkQueue" "Requeueing tab. Requeues: 1"
12:18:17.617 gsUtils.js:41 511 "12:18:17" "checkQueue" "Executing executorFn for tab."
12:18:17.617 gsUtils.js:41 511 "12:18:17" "send message to tab" Object
12:18:17.617 gsUtils.js:41 511 "12:18:17" "response from tab" undefined
12:18:17.618 gsUtils.js:41 511 "12:18:17" "checkQueue" "Reinjecting contentscript into unresponsive unsuspended tab." Object
12:18:17.618 gsUtils.js:41 511 "12:18:17" "Failed to execute js/contentscript.js on tab" Object
12:18:17.618 gsUtils.js:41 511 "12:18:17" "checkQueue" "Queued tab resolved. Result: " "unknown"
12:18:17.618 gsUtils.js:41 checkQueue 12:18:17 total queue size: 0
12:18:17.619 gsUtils.js:41 511 "12:18:17" "Content script status: unknown"
12:18:17.619 gsUtils.js:41 511 "12:18:17" "Focused tab status: loading"
12:18:18.023 gsUtils.js:41 background 12:18:18 updating current session
12:18:18.027 gsUtils.js:41 gsIndexedDb 12:18:18 Updating existing session: 1576514754680
12:18:18.119 gsUtils.js:41 511 "12:18:18" "new stationary tab focus handled"
12:18:19.355 gsUtils.js:41 505 "12:18:19" "tab gained focus"
12:18:19.364 gsUtils.js:41 505 "12:18:19" "send message to tab" Object
12:18:19.367 gsUtils.js:41 505 "12:18:19" "response from tab" Object
12:18:19.367 gsUtils.js:41 505 "12:18:19" "Content script status: formInput"
12:18:19.368 gsUtils.js:41 505 "12:18:19" "Focused tab status: formInput"
12:18:19.627 gsUtils.js:41 508 "12:18:19" "tab gained focus"
12:18:19.635 gsUtils.js:41 508 "12:18:19" "Focused tab status: special"
12:18:20.136 gsUtils.js:41 508 "12:18:20" "new stationary tab focus handled"
12:18:21.226 gsUtils.js:41 511 "12:18:21" "tab gained focus"
12:18:21.228 gsUtils.js:41 511 "12:18:21" "Focused tab status: special"
12:18:21.729 gsUtils.js:41 511 "12:18:21" "new stationary tab focus handled"
12:18:22.757 gsUtils.js:41 -1 "12:18:22" "window gained focus"
</details>

Oddly, I notice that only two of my three windows are showing. One is completely missing even though I know I had it up. Looking back at my sessions, it seems to have been removed after December 12th session, but I know I had it open all those times. I'm not sure why The Great Suspender/Chrome is behaving like this, but I'm more than happy to help you in solving this.

System Information:
Windows 10 Build 1903
Google Chrome Version 79.0.3945.79 (Official Build) (64-bit)
The Great Suspender v7.1.0

For what it's worth, I have the option in Chrome selected to "Continue where you left off" on start up. Though I don't think it matters because I've used the extension for years without this issue with the same setting.

**Comment by q00u** [comment]
I ran into this issue, but I don't use The Great Suspender, and it appears that it might be a Chrome bug.
References:
https://support.google.com/chrome/thread/22667163?hl=en
https://old.reddit.com/r/chrome/comments/e9hvut/psa_chrome_79_has_a_bug_where_minimized_windows/
https://bugs.chromium.org/p/chromium/issues/detail?id=1019048
tl;dr The bug has been fixed upstream, now we're just waiting for the rollout.

**Comment by delusionn** [comment]
This actually is the same issue as 1070.  It's just described differently, as it's easy to encounter it in different places for the first time.  Session Buddy sees the windows, the task bar does not, the Chrome task manager shows lines for them, but if you click them there, you just get a blank window now, etc.

**Comment by q00u** [comment]
_Should_ be fixed in Chrome 79.0.3945.88 stable desktop, which is out now.
https://chromereleases.googleblog.com/2019/12/stable-channel-update-for-desktop_17.html

Installed, relaunched, all my old ghosted windows/tabs have returned.

**Comment by Infinitay** [comment]
Solved, this should be closed too
